examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/MatterCommissioningPrompter.java

```diff
@@ -22,109 +39,103 @@ public class MatterCommissioningPrompter extends UserPrompterResolver implements
   private final String CHANNEL_ID = "MatterCommissioningPrompter.CHANNEL";
   private final int SUCCESS_ID = 0;
   private final int FAIL_ID = 1;
-
+  private final VH mHandler;
   public MatterCommissioningPrompter(Context context) {
     this.context = context;
     this.createNotificationChannel();
-  }
-
-  private Activity getActivity() {
-    return MatterServant.get().getActivity();
+    mHandler = new VH(this);
   }
 
   public void promptForCommissionOkPermission(
-      int vendorId, int productId, String commissioneeName) {
+          int vendorId, int productId, String commissioneeName) {
     // TODO: find app by vendorId and productId
     Log.d(
-        TAG,
-        "Received prompt for OK permission vendor id:"
-            + vendorId
-            + " productId:"
-            + productId
-            + ". Commissionee: "
-            + commissioneeName);
-    AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
-
-    builder
-        .setMessage(commissioneeName + " is requesting permission to cast to this device, approve?")
-        .setTitle("Allow access to " + commissioneeName)
-        .setPositiveButton(
-            "Ok",
-            (dialog, which) -> {
-              OnPromptAccepted();
-            })
-        .setNegativeButton(
-            "Cancel",
-            (dialog, which) -> {
-              OnPromptDeclined();
-            })
-        .create()
-        .show();
+            TAG,
+            "Received prompt for OK permission vendor id:"
+                    + vendorId
+                    + " productId:"
+                    + productId
+                    + ". Commissionee: "
+                    + commissioneeName);
+
+    Message message = Message.obtain();
+    message.arg1 = vendorId;
+    message.arg2 = productId;
+    message.obj = commissioneeName;
```

Comment:
could we encapsulate all data together if we send an object anyway? Why not have a class that contains all of vendorID, productId and name. That way you do not need to remember arg1 being vendor and arg2 being product and naming becomes more safe.